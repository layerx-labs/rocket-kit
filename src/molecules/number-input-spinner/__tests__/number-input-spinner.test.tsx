import React from 'react';
import { render } from '@testing-library/react';
import { NumberInputSpinner } from '../../..';

describe('NumberInputSpinner', () => {
  it('renders', () => {
    const { asFragment } = render(
      <NumberInputSpinner
        decreaseAriaLabel="Decrease Value"
        increaseAriaLabel="Increase Value"
        increment={1}
        min={0}
        max={10}
        value={5}
        onChange={() => {}}
        disabled={false}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
