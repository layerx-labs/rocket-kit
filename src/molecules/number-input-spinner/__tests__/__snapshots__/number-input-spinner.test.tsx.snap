// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NumberInputSpinner renders 1`] = `
<DocumentFragment>
  <div
    class="sc-fFubgz guWIzZ"
  >
    <button
      aria-label="Decrease Value"
      class="sc-idOhPF cKVGdM remove-button"
    >
      <svg
        class="icon"
        viewBox="0 0 32 32"
      >
        <path
          d="M24 17.333h-16c-0.733 0-1.333-0.6-1.333-1.333v0c0-0.733 0.6-1.333 1.333-1.333h16c0.733 0 1.333 0.6 1.333 1.333v0c0 0.733-0.6 1.333-1.333 1.333z"
        />
      </svg>
    </button>
    <input
      class="sc-bkzZxe fPsHLr"
      max="10"
      min="0"
      type="number"
      value="5"
    />
    <button
      aria-label="Increase Value"
      class="sc-idOhPF cKVGdM add-button"
    >
      <svg
        class="icon"
        viewBox="0 0 32 32"
      >
        <path
          d="M24 17.333h-6.667v6.667c0 0.733-0.6 1.333-1.333 1.333s-1.333-0.6-1.333-1.333v-6.667h-6.667c-0.733 0-1.333-0.6-1.333-1.333s0.6-1.333 1.333-1.333h6.667v-6.667c0-0.733 0.6-1.333 1.333-1.333s1.333 0.6 1.333 1.333v6.667h6.667c0.733 0 1.333 0.6 1.333 1.333s-0.6 1.333-1.333 1.333z"
        />
      </svg>
    </button>
  </div>
</DocumentFragment>
`;
